File: DSA/LakshaKundra_590013237/Day_81_Question_3.py
class Node:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None

def print_range(root, L, R):
    if not root:
        return
    if root.val > L:
        print_range(root.left, L, R)
    if L <= root.val <= R:
        print(root.val, end=" ")
    if root.val < R:
        print_range(root.right, L, R)

root = Node(10)
root.left = Node(5)
root.right = Node(15)
root.left.left = Node(3)
root.left.right = Node(7)
root.right.right = Node(18)
print_range(root, 7, 15)
print()

root2 = Node(6)
root2.left = Node(2)
root2.right = Node(8)
root2.left.left = Node(0)
root2.left.right = Node(4)
root2.right.left = Node(7)
root2.right.right = Node(9)
print_range(root2, 2, 8)
print()

File: DSA/VikrantSinghBanwala_590018082/day81_question3.c
#include <stdio.h>
#include <stdlib.h>

typedef struct TreeNode {
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
} TreeNode;

void printNodesInRange(TreeNode* root, int L, int R) {
    if (!root) return;
    if (root->val > L) printNodesInRange(root->left, L, R);
    if (root->val >= L && root->val <= R) printf("%d ", root->val);
    if (root->val < R) printNodesInRange(root->right, L, R);
}

TreeNode* newNode(int val) {
    TreeNode* node = (TreeNode*)malloc(sizeof(TreeNode));
    node->val = val;
    node->left = node->right = NULL;
    return node;
}

int main() {
    TreeNode* root1 = newNode(10);
    root1->left = newNode(5);
    root1->right = newNode(15);
    root1->left->left = newNode(3);
    root1->left->right = newNode(7);
    root1->right->right = newNode(18);
    printNodesInRange(root1, 7, 15);
    printf("\n");

    TreeNode* root2 = newNode(6);
    root2->left = newNode(2);
    root2->right = newNode(8);
    root2->left->left = newNode(0);
    root2->left->right = newNode(4);
    root2->right->left = newNode(7);
    root2->right->right = newNode(9);
    printNodesInRange(root2, 2, 8);
    printf("\n");

    return 0;
}
